### PR TITLE
GitLab CI integration

### PR DESCRIPTION
**Pull request by BirkhoffLee**
This commit integrates Vue.js unit test with GitLab CI hosted at https://gitlab.noj.tw. Also the README was re-written.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# TODO: build & deploy to production via Ansible
+
+stages:
+  - test
+
+unit test:
+  image: node:14.16.1-buster
+  stage: test
+  script:
+    - apt update && apt install -y --no-install-recommends python3
```
Comment (Uier) [inline .gitlab-ci.yml:10]: 在下才疏學淺不知道為什麼此處需要 install python3

**Review by Uier — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (BirkhoffLee) [reply .gitlab-ci.yml:10]: node-gyp dependency

Comment (BirkhoffLee) [reply .gitlab-ci.yml:10]: https://gitlab.noj.tw/noj/frontend/-/jobs/12#L24

**Comment by BirkhoffLee** [comment]
Per the commit message of 66b0d52:

> This commit introduces the integration to the internal GitLab CI
system.
> 
> 1. Removes the existing "testing" code, where they are just examples
of the testing suite.
> 2. Dockerfile is now a Caddy container that serves the built static
   files (built by GitLab CI)
> 3. .gitlab-ci.yml includes the unit testing code, so that we can easily
   integrate in the future
> 4. After the Dockerfile gets built, it's uploaded to GitLab CI Container
   Registry.
> 
> TODO:
> 1. Deploy to production/staging server with Ansible.
> 2. Deployments for Merge Requests.
> 
> Signed-off-by: Birkhoff Lee <git@birkhoff.me>

The CI pipeline has run successfully at GitLab. This commit is now ready to be merged into master.

**Pull request merged**